packages/esm-form-entry-app/src/app/form-data-source/form-data-source.service.ts

```diff
@@ -114,6 +117,56 @@ export class FormDataSourceService {
     );
   }
 
+  public getMostRecentObsValueBefore(date: string, patientUuid: string) {
+    type ObsResult = FHIRResource['resource'] & {
```

Comment:
While this isn't per-se harmful, type declarations should generally not be declared in functions. There's already a `types.ts` file that we're importing from... Why not declare this type there?